### DISCUSSION TITLE
Sveltekit-i18n download all the language.json file

### DISCUSSION THREAD

**Discussion by VisionaryAppDev** [sveltekit-i18n/lib#141] [Q&A] [answered]
I have found this lib today and of course it's one of the cleanest and simplest lib. But it seems like it does load all the translation.json files even the page doesn't need those translation. So, isn't that will affect the page performance and consume a lot of bandwidth especially on a large project?

**Comment by jarda-svoboda** [comment, marked as answer]
Hi @VisionaryAppDev, thanks for your question.
The load mechanism is quite simple:
1) Your config can include [`translations`](https://github.com/sveltekit-i18n/lib/tree/master/docs#translations-translationst) object, serialized by locale, which is loaded instantly.
2) [`loaders`](https://github.com/sveltekit-i18n/lib/tree/master/docs#loaders-loaderloadermodule) array, defining which translation definitions (jsons) should be loaded for which `routes` and `locales`.
– when no `routes` are defined within a loader, then it's loaded together with the first page.

I think [this example](https://github.com/sveltekit-i18n/lib/tree/master/examples/loaders) might help you to understand `loaders` better.. (see the `loaders` config [here](https://github.com/sveltekit-i18n/lib/blob/master/examples/loaders/src/lib/translations/index.js#L20))

**Reply by VisionaryAppDev** [reply]
Thank @jarda-svoboda for your answer. It's great that all of these features are already available! Can't wait to try it out!

[a comment by an automated account is not shown]
